cmd/agent/main.go

```diff
@@ -47,6 +47,17 @@ func main() {
 		if cfg != nil {
 			cfg.Server.Log = cfgLogger
 		}
+		// Iterate over the configurations and remove any that Enabled = false, this works by setting i to the last element
+		//   and then returning the array minus the last element. This loses order but is performant.
+		for i := 0; i < len(cfg.Integrations.Integrations);{
+			if !cfg.Integrations.Integrations[i].CommonConfig().Enabled {
+				cfg.Integrations.Integrations[i] = cfg.Integrations.Integrations[len(cfg.Integrations.Integrations)-1]
+				cfg.Integrations.Integrations = cfg.Integrations.Integrations[:len(cfg.Integrations.Integrations)-1]
+			} else {
+				i++
+			}
+		}
```

Comment:
Can you create a new function and call it from integrations.Manager.ApplyConfig instead? Filtering the actual slice here will omit the config from the results of `/-/config` which might be a little confusing to users.

Reply:
This got more complicated. When you remove a configuration it did not stop the scraper, so when a configuration is disabled/removed and reload called Agent would list scrape_error = 404. Since the integration had stopped but the scraper was still running.